RemoveStatusCondition panics with an empty slice

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


#### What happened:

Attempting to remove a status condition from an already empty slice panics.

```plaintext
panic: runtime error: makeslice: cap out of range

goroutine 1 [running]:
k8s.io/apimachinery/pkg/api/meta.RemoveStatusCondition(0xc000157f60, 0x93ec68, 0x3)
        /home/mark/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/api/meta/conditions.go:63 +0x73
main.main()
        /home/mark/src/testing/main.go:10 +0x52
exit status 2
```

#### What you expected to happen:

An empty slice should be removed (i.e. removed idempotently) and without panicking.

#### How to reproduce it (as minimally and precisely as possible):

```go
package main

import (
	"k8s.io/apimachinery/pkg/api/meta"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func main() {
	conditions := []metav1.Condition{}
	meta.RemoveStatusCondition(&conditions, "Foo")
}
```

#### Anything else we need to know?:

Originally found with `k8s.io/apimachinery@v0.20.1`, reproduced with `k8s.io/apimachinery@latest` (`v0.20.4` at the time of writing)

#### Environment:
- Kubernetes version (use `kubectl version`): N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g: `cat /etc/os-release`): Pop!_OS 20.04 LTS
- Kernel (e.g. `uname -a`): `Linux mark-l 5.8.0-7642-generic #47~1612288990~20.04~b8113e7-Ubuntu SMP Wed Feb 3 02:25:36 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools: N/A
- Network plugin and version (if this is a network-related bug): N/A
- Others: N/A

--- a/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go
+++ b/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go
@@ -57,7 +57,7 @@ func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Cond
 // RemoveStatusCondition removes the corresponding conditionType from conditions.
 // conditions must be non-nil.
 func RemoveStatusCondition(conditions *[]metav1.Condition, conditionType string) {
-	if conditions == nil {
+	if conditions == nil || len(*conditions) == 0 {
 		return
 	}
 	newConditions := make([]metav1.Condition, 0, len(*conditions)-1)

--- a/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions_test.go
+++ b/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions_test.go
@@ -121,6 +121,12 @@ func TestRemoveStatusCondition(t *testing.T) {
 				{Type: "third"},
 			},
 		},
+		{
+			name:          "empty_conditions",
+			conditions:    []metav1.Condition{},
+			conditionType: "Foo",
+			expected:      []metav1.Condition{},
+		},
 	}
 
 	for _, test := range tests {

